### PR TITLE
Adjust EROS adapter to match new integration spec

### PR DESCRIPTION
**Pull request by chelseatroy**
Theia is designed to be a one stop shop for researchers to request images from NASA LANDSAT satellites (and someday possibly other satellites), perform manipulations on those images (like resizing, cropping, or colorizing them), and upload those to a project on panoptes to use as subject sets.

As such, theia integrates with a variety of services. One of the very first services with which it integrates is EROS.

EROS is the United States Geological Survey's API for searching for images from the LANDSAT satellites. We integrate with EROS to allow researchers to search for, and request, the images they need for their projects. This involves two steps:

1. Authenticating with EROS
2. Making the imagery requests themselves

Once upon a time, the way EROS authenticated users was by having clients stick their credentials into a JSON request object that got URL encoded and jammed onto the end of the request URL.

We suspect that, at some point, someone over at EROS saw that and told the team that it was bad practice to do that. So now, for authentication, EROS does not accept that. Instead, it expects the credentials in the body of the authentication request.

Now, for the _image search_ requests, EROS does _not_ accept the credentials in the body of the request. These requests still expect credential information jammed onto the request URL. 

So this PR changes our wrapper such that the authentication request submits credentials in the body, and the other requests submit credentials in the request URL. 

It also locks our integration to version 1.3.0 of the EROS API so that changes to the API don't break us anymore.

To test:
1. Navigate to app directory
1. `$ pipenv run console`
1. `$ from theia.tasks import locate_scenes`
1. Go into the Admin UI or the REST UI and create an imagery request. Get the ID of that imagery request.
1. `$ locate_scenes(<ImageryRequest id>)`

If the integration fails, you'll see the error from EROS in the console there. If it appears to hang for a while and then returns nothing, it worked.

### CODE REVIEW

File: theia/adapters/usgs/eros_wrapper.py [hunk 1]
```diff
@@ -62,40 +62,52 @@ def parse_result_set(cls, result_set):
         return [scene.get('displayId', None) for scene in result_set if 'displayId' in scene]
 
     @classmethod
-    def eros_get(cls, url, request_data, **kwargs):
+    def eros_get(cls, url, request_data, authenticating=False, **kwargs):
         if url != 'login' and (not cls.token()):
             cls.connect()
 
-        new_args = cls.eros_prepare(request_data, **kwargs)
+        new_args = cls.eros_prepare(request_data, authenticating=authenticating, **kwargs)
```
Comment (zwolf) [inline theia/adapters/usgs/eros_wrapper.py]: non-blocking: this is a little awkward, might change up this var name

File: theia/adapters/usgs/eros_wrapper.py [hunk 2]
```diff
@@ -62,40 +62,52 @@ def parse_result_set(cls, result_set):
...
-        }
+        if authenticating:
+            headers['Content-Type'] = 'application/x-www-form-urlencoded'
+
+            if request_data:
+                data = 'jsonRequest=' + quote(json.dumps(request_data))
+
+            new_args = {
+                'headers': headers,
+                'data': data,
```
Comment (zwolf) [inline theia/adapters/usgs/eros_wrapper.py]: Doesn't look like this is instantiated anywhere if no request data is provided like `params` is up there.

Comment (chelseatroy) [reply theia/adapters/usgs/eros_wrapper.py]: Yeah, a few of these things I'm not quite sure about 🤔still trying to figure out exactly how this app works and which things need to be there for "django magic" to occur.

File: theia/adapters/usgs/eros_wrapper.py (hunk 1 above)
Comment (chelseatroy) [reply theia/adapters/usgs/eros_wrapper.py]: I have some changes coming up to the eros wrapper. You raise a good point. Will change in those changes!

**Pull request merged**